src/pages/resultsView/pathwayMapper/ResultsViewPathwayMapper.tsx

```diff
@@ -0,0 +1,222 @@
+import * as React from "react";
+import * as _ from "lodash";
+import $ from "jquery";
+import { QueryParameter } from "../../../shared/lib/ExtendedRouterStore";
+import {ResultsViewPageStore, SamplesSpecificationElement, IQueriedMergedTrackCaseData, IQueriedCaseData} from "../ResultsViewPageStore";
+import { ResultsViewTab } from "../ResultsViewPageHelpers";
+import { generateGeneAlterationData } from "../download/DownloadUtils";
+import PathwayMapper, {ICBioData} from "react-pathway-mapper";
+import "react-pathway-mapper/dist/base.css";
+import PathwayMapperTable, { IPathwayMapperTable } from "./PathwayMapperTable";
+import { observer } from "mobx-react";
+import autobind from "autobind-decorator";
+import { observable, ObservableMap, computed } from "mobx";
+import { alterationInfoForOncoprintTrackData } from "shared/components/oncoprint/OncoprintUtils";
+import { isMergedTrackFilter } from "shared/lib/oql/oqlfilter";
+import { Sample, Patient, MolecularProfile, Gene } from "shared/api/generated/CBioPortalAPI";
+import { makeGeneticTrackData } from "shared/components/oncoprint/DataUtils";
+import { CoverageInformation } from "../ResultsViewPageStoreUtils";
+import { Grid, Col, Row } from "react-bootstrap";
+
+import { AppStore } from "AppStore";
+import { remoteData } from "cbioportal-frontend-commons";
+import LoadingIndicator from "shared/components/loadingIndicator/LoadingIndicator";
+import { fetchGenes } from "shared/lib/StoreUtils";
+import { ErrorMessages } from "shared/enums/ErrorEnums";
+import OqlStatusBanner from "shared/components/banners/OqlStatusBanner";
+
+import 'react-toastify/dist/ReactToastify.css';
+import {ToastContainer, toast} from 'react-toastify';
+
+interface IResultsViewPathwayMapperProps{
+    store: ResultsViewPageStore;
+    initStore: Function;
+    appStore: AppStore;
+}
+
+@observer
+export default class ResultsViewPathwayMapper extends React.Component<IResultsViewPathwayMapperProps, {}>{
+    
+
+    @observable
+    tableData: IPathwayMapperTable[];
+
+    cBioData: ICBioData[];
+
+    @observable storeForAllData: ResultsViewPageStore;
+
+    @observable selectedPathway: string = "";
+
+    pathwayHandler: Function;
+
+    @observable
+    isLoading: boolean;
+
+    @observable
+    currentGenes: string[];
+
+    @observable
+    validGenes: {[gene: string] : boolean};
```

Comment:
This can be a `@computed`, so that you don't need to update it in `onResult` method of a `remoteData`. Instead, you can access `remoteGenes` inside your `@computed get validGenes()`.